src/Interpreters/Streaming/Aggregator.h

```diff
@@ -0,0 +1,1014 @@
+#pragma once
```

Comment:
We choose to copy and revise the Aggregator because we like to avoid impacting the efficiency / performantce for historical data processing. And aggregator will have these major different tweaks / behaviors / revisions for now and moving forward.

1. Keep around states after intermediate aggregation emit. (We have this in the current PR)
2. Garbage collect aggregate states when watermark progress for streaming window process (in future PR)
3. Interface changes -> Block => Chunk to make it more lightweight / efficient for streaming processing.
4. State checkpointing and restore. In streaming processing, when enable checkpoint, we will need checkpoint the in-memory state and restore it when the query gets recovered.